### PR TITLE
Merge: Switch to Socket.IO Networking Solution

### PR DESCRIPTION
**Pull request by timkanbur**
Due to the synchronous and unfixable previous python-socket-networking scripts, we should transition to Socket.IO to resolve previous networking limitations.

### CODE REVIEW

File: sio_client.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+        async def any_event(event, data):
+            """Handles any event and triggers the corresponding event."""
+            logging.info(f"[Net-Any] ({event}): {data}")
+            await self.event_manager.trigger_event(event, data)
+
+
+    async def connect(self):
+        """Attempts to connect to the server and returns the success status."""
+        try:
+            await self.sio.connect(f'http://localhost:{self.server_port}')
```
Comment (DarkS0ul3) [inline sio_client.py:72]: I'm not sure if localhost is the right way for the connection. Like, if i want to connect to another client, i normally have to specify the client ip (or at least the domain of the client, which should probably not be localhost?)

File: sio_client.py [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+            logging.error(f'[Net-SendData] Fehler: {e}')
+
+    
+    async def discover(self):
+        """Discovers potential servers in the local network."""
+        try:
+            self.potential_servers = []
+            for port in range(self.SERVER_MIN_PORT, self.SERVER_MAX_PORT + 1):
+                try:
+                    await self.sio.connect(f'http://localhost:{port}')
```
Comment (DarkS0ul3) [inline sio_client.py:93]: Same here. Even if there is only one client in the network, it should have a different domain than localhost (ip even better)

**Review by DarkS0ul3 — COMMENTED** [review]
There might be some changes to do regarding the connect()- and discover()-function

File: sio_client.py (hunk 1 above)
Comment (timkanbur) [reply sio_client.py:72]: Yes, at the moment the connection is only running locally for testing and implementation purposes. Once the guidelines and UI elements are implemented, a quick addition should  change this very easily.

Comment (DarkS0ul3) [reply sio_client.py:72]: Alright 👍

**Pull request closed without merging**